parser.go

```diff
@@ -138,3 +148,83 @@ func (p *Parser[V]) Pop() *Node[V] {
 	p.node = p.node.Parent
 	return n
 }
+
+// RotateLeft moves the current node to the position it's parent had,
+// making the parent into the current node's child.
```

Comment:
nit: Add a note about what happens to the parent's other children.

Reply:
Pushed.